src/sys/petsclib.cxx

```diff
@@ -22,6 +22,24 @@ PetscLib::PetscLib() {
     PetscInitialize(pargc,pargv,PETSC_NULL,help);
     PetscLogEventRegister("Total BOUT++",0,&USER_EVENT);
     PetscLogEventBegin(USER_EVENT,0,0,0,0);
+
+    // Load global PETSc options from the [petsc] section of the input
+    setPetscOptions(Options::root()["petsc"], "");
+  }
+
+  if (opt != nullptr and opt->isSection("petsc")) {
+    // Use options specific to this PetscLib
+    // Pass options to PETSc's global options database, with a unique prefix, that will be
+    // passed to a KSP later.
+    // (PetscOptions type exists for non-global options, but apparently is only for user
+    // options, and cannot be passed to KSP, etc. Non-global options can be passed by
+    // defining a custom prefix for the options string, and then passing that to the KSP.)
+
+    options_prefix = "boutpetsclib" + std::to_string(count) + "_";
+
+    Options& options = (*opt)["petsc"];
```

Comment:
Rather than take the "petsc" subsection from `opt`, why not just use `opt` itself, and pass in the correct subsection when constructing `PetscLib`? I'm not sure we're terribly consistent with this, but I think it's clearer that when we take an `Options` we will use that section itself, and not look for a particular subsection.

Reply:
I've done it this way because if the `petsc` sub-section isn't in `opt` (see l.30), then `options_prefix` is left as `""` and the solver will use the global options from the `[petsc]` top-level section and/or the command line.

Reply:
Sure, but if e.g. `LaplaceXY` passed `Options::root()["laplacexy"]["petsc"]` and L30 was

```c++
 if (opt != nullptr and opt->isSection()) {
```

that would be equivalent?

Reply:
I guess that might work. In the case where `petsc` is not there, would your way create a new entry in the options tree? Maybe it doesn't matter if it did?